### PR TITLE
When reading and writing the video CSV file, take locale into account

### PR DESCRIPTION
**Pull request by bethanyaconnor**
- Removes key uniqueness requirement in the CSV
- Pull out the function calls to modify the translation file and the CSV to a helper function
- Only translate English titles
  - The way this is currently designed/implemented will mean that we don't save non-English titles
- This code changes relies on PR #27152 to have been applied, so this change will be merge after that change has gone out in at least one DTP.

### CODE REVIEW

File: dashboard/app/controllers/videos_controller.rb [hunk 1]
```diff
@@ -117,4 +115,11 @@ def i18n_params
   prepend_before_action do
     params[:video] &&= video_params
   end
+
+  def merge_and_write
+    if @video.locale == 'en-US'
```
Comment (joshlory) [inline dashboard/app/controllers/videos_controller.rb:120]: Does the view also need to be updated to pass a locale, so current video saves continue to work in the interim?

(might be good to add a test to cover this case, if there isn't one)

Comment (Hamms) [reply dashboard/app/controllers/videos_controller.rb:120]: :+1: for tests!

Comment (bethanyaconnor) [reply dashboard/app/controllers/videos_controller.rb:120]: Added a test for what is passed into merge_and_write_attributes_stub. As 'en-US' is the default locale in Video, this "works" without a view change.

Some changes in the levelbuilder create video view will be sent out in a separate PR.

Comment (joshlory) [reply dashboard/app/controllers/videos_controller.rb:120]: Bethany confirmed the schema.rb default gets loaded by Video.new, TIL!


image

**Review by joshlory — APPROVED** [review]
LGTM!

**Pull request merged**